File: 2_Programação_sistemas_operacionais/2.1_File_stdio/Python/Ex5.py
def abre_arq(arquivo, modo):
	try:
		p = open(arquivo, modo)
	except OSError:
		print("Erro! Impossivel abrir o arquivo!")
		exit(-1)
	return p

def le_arq(p):
	c = fp.read(1000)
	while c != '':
		print(c, end='')
		c = fp.read(1000)

frase = "Este é um arquivo chamado: "
print("Entre com um nome para o arquivo:")
str = input()
fp = abre_arq(str, "w");
frase += str + "\n"
fp.write(frase)
fp.close()

fp = abre_arq(str, "r");
le_arq(fp)
fp.close()

File: 2_Programação_sistemas_operacionais/2.1_File_stdio/Ex5.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "file_stdio.h"

int main(int argc, const char * argv[])
{
	FILE *p;
	char str[100], c;
	char frase[120] = "Este e um arquivo chamado: ";
	printf("Entre com um nome para o arquivo:\n");
	gets(str);
	p = abre_arq(str, "w");
	strcat(frase, str);
	fputs(frase, p);
	putc('\n', p);
	fclose(p);
	p = abre_arq(str, "r");
	le_arq(p);
	fclose(p);
	return 0;
}
